src/api/app/controllers/webui/package_controller.rb

```diff
@@ -1021,6 +1021,18 @@ def save_meta
 
   def edit; end
 
+  def binary_download
+    filename = File.basename(params[:filename]) # Ensure it really is just a file name, no '/..', etc.
+    repository = Repository.find_by_project_and_name(@project.to_s, params[:repository].to_s)
+
+    download_url = download_url_for_file_in_repo(@project, @package, repository, params[:arch], filename)
+    if download_url
```

Comment:
This can be nil for two reasons, the file not being [available](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/controllers/webui/package_controller.rb#L1125) and the user not being [logged in](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/controllers/webui/package_controller.rb#L1127).
For the latter case I would first check if the user is logged in and only then check for the file, etc.